[WIP] [DO NOT MERGE] RHDEVDOCS-5369: Pipelines as Code, Logging Configuration (#63525)
openshift/openshift-docs: modules/op-configuring-pipelines-as-code-logging.adoc

// This module is included in the following assembly:
//
// *cicd/pipelines/using-pipelines-as-code.adoc

:_content-type: PROCEDURE
[id="configuring-pipelines-as-code-logging_{context}"]
= Configuring {pac} logging 

> You can configure {pac} logging with the `pac-config-logging` config map located in the same namespace as the controllers. The default namespace is `openshift-pipelines`.

Reviewer: IMHO since it's not possible to configure namespace in downstream, we could merge those two sentences `... config map located in the namespaces openshift-pipelines`.

metadata:
  labels:
    app.kubernetes.io/instance: default
    app.kubernetes.io/part-of: pipelines-as-code
  name: pac-config-logging
  namespace: openshift-pipelines
data:
  loglevel.pac-watcher: info <1>
  loglevel.pipelines-as-code-webhook: info <2>
  loglevel.pipelinesascode: info <3>
  zap-logger-config: |
    {
      "level": "info",
      "development": false,
      "sampling": {
        "initial": 100,
        "thereafter": 100
      },
      "outputPaths": ["stdout"],
      "errorOutputPaths": ["stderr"],
      "encoding": "json",
      "encoderConfig": {
        "timeKey": "ts",
        "levelKey": "level",
        "nameKey": "logger",
        "callerKey": "caller",
        "messageKey": "msg",
        "stacktraceKey": "stacktrace",
        "lineEnding": "",
        "levelEncoder": "",
        "timeEncoder": "iso8601",
        "durationEncoder": "",
        "callerEncoder": ""
      }
    }
----
<1> The log level for the `pipelines-as-code-watcher` component
<2> The log level for the `pipelines-as-code-webhook` component
<3> The log level for the `pipelines-as-code-controller` component

. Change the log level to any supported value:
+
[source,terminal]
----

> $ oc patch configmap pac-config-logging -n openshift-pipelines --type json -p '[{"op": "replace", "path": "/data/loglevel.<component>", "value":"<value>"}]'

Reviewer 2: @piyush-garg @savitaashture does this work ? don't we require users to update the tektonconfig object instead ?

Reviewer 3: Yes because we don't update cm data using tektonconfig this is my understanding

Reviewer 2: ok then 👼🏼

Author: @piyush-garg is that also true for Pipelines 1.12? We got informed before that the user will be able to modify this configmap directly in 1.12, that is why this issue exists for 1.12.

Reviewer 4: i checked this, the value is getting reverted back in the configmap of the installerset, but not on the cluster. That seems like some other bug. But his will be reverted in case of upgrade atleast. So this approach might not work

Author: Hello @vdemeester What does this mean for the issue? Will it be accessible for the future versions or is this approach not planned? Because in that case, I would close this PR and close the issue as "won't do" or something similar.

Reviewer 2: I think we need to document this (the behavior @piyush-garg highlighted) as a known issues.
@piyush-garg @savitaashture do we have an issue for it ?

Reviewer 5: @vdemeester @piyush-garg @savitaashture question: can the log levels, and/or other settings in this configmap, be configured using TektonConfig? If they can. this is the preferred option anyway. But it they can not be configured by TektonConfig while configuration in the configmap gets overwritten, then it should be a known issue - and a corresponding bug Jira should be opened so Dev can provide the proper way of configuring logging.

Reviewer 2: I think they cannot be configured through `TektonConfig` today, hence creating an issue and documenting that as a known issue or something.

Reviewer 4: if it is only about log level of three components, maybe we can add them as part of spec.

// This module is included in the following assembly:
//

> // *cicd/pipelines/using-pipelines-as-code.adoc

Reviewer 6: ```suggestion
// * cicd/pipelines/using-pipelines-as-code.adoc
```

:_content-type: PROCEDURE
[id="configuring-pipelines-as-code-logging_{context}"]
= Configuring {pac} logging 

You can configure {pac} logging by editing the `pac-config-logging` config map in the `TektonConfig` custom resource (CR).

.Prerequisites

> * {pac} is installed on your cluster.

Reviewer 6: ```suggestion
* You have {pac} installed on your cluster.
```

.Procedure

> . In the *Administrator* perspective of the web console, go to *Administration* → *CustomResourceDefinitions*.

Reviewer 6: Just with consistency on web console docs
```suggestion
. In the *Administrator* perspective of the web console, navigate to *Administration* → *CustomResourceDefinitions*.
```

Author: based on the IBM style guide: 
**navigate** Use only when a more direct verb, such as "click", "select", "browse", or "go to", is not applicable. Use only to provide direction to a location (for example, "Navigate to the directory that contains the files that you want to edit").

So I will keep "go" instead.

Reviewer 6: If you choose to use "go" that is up to you like you mentioned the style guide does say you can use go; however, with the guideline "Use only to provide direction to a location" this would be a direction to a location with the final location being *CustomResourceDefinitions*. As I said, it is up to you but wanted to explain my reasoning further.

> . Use the *Search by name* box to search for the `tektonconfigs.operator.tekton.dev` custom resource definition (CRD) and click *TektonConfig* to see the CRD details page.

Reviewer 6: I think we should use field instead of box here based on ISG https://www.ibm.com/docs/en/ibm-style?topic=elements-ui#field

> . Use the *Search by name* box to search for the `tektonconfigs.operator.tekton.dev` custom resource definition (CRD) and click *TektonConfig* to see the CRD details page.

Reviewer 6: ```suggestion
. Use the *Search by name* box to search for the `tektonconfigs.operator.tekton.dev` custom resource definition (CRD) and click *TektonConfig* to see the CRD *Details* page.
```
I think. I don't have a cluster up right now to see the UI, but the name of the page should be bolded.

Author: I'll check and make the appropriate change, thank you!

Author: i just checked, it should be *Details*, thank you for noticing!

. Click the *Instances* tab.

> . Click the *config* instance to see the `TektonConfig` CR details.

Reviewer 6: What is the instance? Is it a link that shows a pop up window? If so, I think it should be
```suggestion
. Click *config*  to view the `TektonConfig` CR details in a pop-up window.
```
See https://redhat-documentation.github.io/supplementary-style-guide/#popup for more

Author: I am pretty sure there is not a pop-up window, when the user clicks on the *config* instance, but I will double check.

Author: The instance is the config. That is the name for that particular instance.

. Click the *YAML* tab.

. Edit the `loglevel.` fields under the `.options.configMaps.pac-config-logging.data` parameter based on your requirements.
+

> .Example: `TektonConfig` CR with {pac} log level fields set to `warn`

Reviewer 6: I have never seen this phrased this way. I am wondering if this might work better? 
```suggestion
.Sample YAML of `TektonConfig` CR with {pac} log level fields set to `warn`
```

Author: @opayne1 We actually had this discussion before in the team-ocp-docs. In Pipelines documentation we use the "Example: Description" format almost every time so I wanted it to stay consistent with the rest of the documentation. But the "Example of <description.>" is also occasionally used.

            pac-config-logging:
              data:
                loglevel.pac-watcher: warn <1>
                loglevel.pipelines-as-code-webhook: warn <2>
                loglevel.pipelinesascode: warn <3>
                zap-logger-config: |
                  {
                    "level": "info",
                    "development": false,
                    "sampling": {
                      "initial": 100,
                      "thereafter": 100
                    },
                    "outputPaths": ["stdout"],
                    "errorOutputPaths": ["stderr"],
                    "encoding": "json",
                    "encoderConfig": {
                      "timeKey": "ts",
                      "levelKey": "level",
                      "nameKey": "logger",
                      "callerKey": "caller",
                      "messageKey": "msg",
                      "stacktraceKey": "stacktrace",
                      "lineEnding": "",
                      "levelEncoder": "",
                      "timeEncoder": "iso8601",
                      "durationEncoder": "",
                      "callerEncoder": ""
                    }
                  }
----

> <1> The log level for the `pipelines-as-code-watcher` component

Reviewer 6: To remain consistent with the rest of the docs, each of these need a period.

<2> The log level for the `pipelines-as-code-webhook` component
<3> The log level for the `pipelines-as-code-controller` component

> . Optional: Create custom logging config map for the {pac} components by changing the `.env.value` for each component under `.options.deployments` field. The example below shows the configuration with the custom config map called `custom-pac-config-logging`.

Reviewer 6: ```suggestion
. Optional: Create custom logging config map for the {pac} components by changing the `.env.value` for each component under the `.options.deployments` field. The example below shows the configuration with the custom config map for `custom-pac-config-logging`.
```
I think "called" should be "for" here though I dont think there is anything against using "called" in the style guides.

Author: Because here I talk about the custom config map called `custom-pac-config-logging`, I think changing it to "for" changes the meaning of the sentence. With "for" it sounds like the example shows some `custom-pac-config-logging` object that contains custom config map, which is wrong, because `custom-pac-config-logging` is the config map.

+

> .Example: `TektonConfig` CR with {pac} custom logging config map

Reviewer 6: Same with this one here
```suggestion
.Sample YAML of `TektonConfig` CR with {pac} custom logging config map
```
